probe-rs/src/flashing/flasher.rs

```diff
@@ -599,6 +610,28 @@ impl<'probe, O: Operation> ActiveFlasher<'probe, O> {
         Ok(())
     }
 
+    fn check_for_stack_overflow(&mut self) -> Result<(), FlashError> {
+        let algo = &self.flash_algorithm;
+
+        if !algo.stack_overflow_check {
+            return Ok(());
+        }
+
+        let stack_bottom = algo.stack_top - algo.stack_size;
+        let read_back = self
+            .core
+            .read_word_8(stack_bottom)
+            .map_err(FlashError::Core)?;
+
+        if read_back != STACK_FILL_BYTE {
```

Comment:
There are at least two ways to improve this:
 - check more bytes
 - use random fill data
But hopefully this solution still works better than no checks at all :)